server/gitrest/packages/gitrest-base/src/utils/filesystems.ts

```diff
@@ -3,11 +3,18 @@
  * Licensed under the MIT License.
  */
 
-import fs from "fs";
+import fs from "node:fs";
+import { Volume } from "memfs";
 import { IFileSystemManager, IFileSystemManagerFactory, IFileSystemManagerParams } from "./definitions";
 
 export class NodeFsManagerFactory implements IFileSystemManagerFactory {
     public create(params?: IFileSystemManagerParams): IFileSystemManager {
         return fs;
     }
 }
+
+export class MemFsManagerFactory implements IFileSystemManagerFactory {
+    public create(params?: IFileSystemManagerParams): IFileSystemManager {
+        return (new Volume() as unknown) as IFileSystemManager;
```

Comment:
Hey Xin, this is only used very briefly in 1 function call and should be picked up by GC like any other variable. It is not stored external to the function, and is identical to the logic in Spork, which had a memory leak because of a global object